metricbeat/mb/registry.go

```diff
@@ -105,6 +107,16 @@ type Register struct {
 	modules map[string]ModuleFactory
 	// A map of module name to nested map of MetricSet name to MetricSetRegistration.
 	metricSets map[string]map[string]MetricSetRegistration
+	// Child registry that can contain additional modules
+	child ChildRegister
+}
+
+// ChildRegister contains an additional source of modules
+type ChildRegister interface {
```

Comment:
In Go there's not _parent_-_child_ relationships. You have _composed_ or _part of_. It's not a subtle difference because in OOP a child has all the properties of a parent

Reply:
By the way, why not a `RegisterInterface` (child or not) with Strategies like a `ChildRegister struct`, `HttpRegister struct`, etc. which allows many types of different types of registerers?

Reply:
I have changed this to `ModuleSource`, as this defines better what they are. The light modules implementation is then `LightModulesSource`, and current `Register` could implement this interface if we decide to refactor it at some moment to follow the same interface. If we would have modules from http endpoints we could implement a `HttpModulesSource` or similar.